src/GslibIO.jl

```diff
@@ -52,28 +81,46 @@ Optionally set the value used for missings `na`.
 """
 function load_legacy(filename::AbstractString, dims::NTuple{3,Int};
                      origin=(0.,0.,0.), spacing=(1.,1.,1.), na=-999)
-  open(filename) do fs
-    # skip header
-    readline(fs)
+  spec = parse_legacy(filename)                     
 
-    # number of properties
-    nvars = parse.(Int, readline(fs))
+  # handle missing values
+  replace!(spec.data, na=>NaN)
 
-    # properties names
-    vars = [Symbol(readline(fs)) for i in 1:nvars]
+  # create data dictionary
+  table = (; zip(spec.varnames, eachcol(spec.data))...)
+  domain = RegularGrid(dims, origin, spacing)
 
-    # read property values
-    X = readdlm(fs)
+  georef(table, domain)
+end
```

Comment:
Perfect, I think we have the regular grid case workign now and with very clear code. 👍🏽